src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -364,12 +371,17 @@ public void setWhitelisted(boolean value)
 	@Override
 	public Player getPlayer()
 	{
-		if (online)
-		{
-			return this;
-		}
+		return (online) ? this : null;
+	}
 
-		return null;
+	/**
+	 * Sets the return value of {@link #isOnline}.
+	 *
+	 * @param online Whether the player is online.
+	 */
+	public void setOnline(boolean online)
+	{
+		this.online = online;
 	}
 
```

Comment:
Please remove this setter.
This would cause a lot of problems as it does not actually affect the online status of the player. If a player were to (artifically) disconnect from the server and someone forgets to call this method, it would leave it in an inconsistent state.
Likewise, if someone were to manipulate the internal state using this method, it would create another inconsistency.